### DISCUSSION TITLE
NHS.UK frontend v7.0.0 alpha discussion

### DISCUSSION THREAD

**Discussion by wa-rren-dev** [nhsuk/nhsuk-frontend#827] [General]
Any comments or issues with v7 alpha of nhsuk-frontend welcomed here.

Topics discussed:
- [What does `alpha` mean?](https://github.com/nhsuk/nhsuk-frontend/discussions/827#discussioncomment-3157183)
- [Should we use Prettier?](https://github.com/nhsuk/nhsuk-frontend/discussions/827#discussioncomment-3188144) - Pull down https://github.com/nhsuk/nhsuk-frontend/pull/832 to try the current suggestion!
- [Is it time to move away from Gulp?](https://github.com/nhsuk/nhsuk-frontend/discussions/827#discussioncomment-3189236)

**Comment by andymantell** [comment]
Thanks for getting the ball rolling on this stuff.

I think it would be useful to know what the criteria is for stuff to be in alpha. How does it differ to that on the main branch? And what's the criteria to move from alpha to main?

I think we need to be clear on this so that if I, or others, decide to install the alpha version on their services, we know what we're in for?

**Reply by wa-rren-dev** [reply]
Hi Andy, we've got no official line but in my mind it's a testing ground for anything really. The breaking change just comes from the removal of support for the older sass compilers. Dom's [beta version](https://github.com/nhsuk/nhsuk-frontend/tree/beta) is very much tailored to his team. Next week I'd like to get your [tabs PR](https://github.com/nhsuk/nhsuk-frontend/pull/750) in there so whoever wants to can start playing.

I suppose the criteria would be that ideally there's supporting information for any new components in the service manual. So when we're ready for the service manual to use v7 that's when we merge it to main.

**Reply by andymantell** [reply]
Sounds good to me - requiring service manual criteria for main but not alpha makes a lot of sense.

**Reply by pflynny** [reply]
It would be great to get this working alongside the beta branch, as it was never intended for that to become specific for the navigation team's components; it was more just an opportunity to remove things from being siloed in the CMS.

**Comment by wa-rren-dev** [comment]
Welcome any thoughts on whether we move the linting / formatting to Prettier....?

**Reply by andymantell** [reply]
We just had this conversation, but I'll weigh in here as well. I can't work without Prettier any more, it's invaluable. So I would vote strongly to move to a Prettier based setup which fixes things as you go, instead of a pre-commit hook that punishes you for mistakes.

**Reply by andymantell** [reply]
You still need eslint as it still covers some things that Prettier does not. It's really about making the basic eslint rules match prettier. I've got a setup I can share which works

**Reply by wa-rren-dev** [reply]
Thanks @andymantell could you share that?

**Reply by andymantell** [reply]
@wa-rren-dev Sure, I think this includes all the relevant bits:
https://gist.github.com/andymantell/058288dc66f657268e332cfc2dd6b136

**Reply by pflynny** [reply]
I agree with @andymantell's comments, it improves workflow if it is monitored during dev instead of pre-commit, so I'd back a switch to prettier.

**Reply by wa-rren-dev** [reply]
Just for visibility in this chat - there's a PR that can be played with at https://github.com/nhsuk/nhsuk-frontend/pull/832

**Comment by pflynny** [comment]
Is it time to move on from Gulp? 

Gulp hasn't had a major release since 2019 and has very little activity on the repo. Some projects had pipelines failing due to an issue/security and I believe they created their own webpack build task.

**Reply by andymantell** [reply]
111 online is in the process of ditching gulp - we have moved to a fully webpack based solution with a handful of npm scripts for shuffling files around if necessary (Things like copying some images out of the nhsuk-frontend package into the right place). I haven't found anything so far that I need gulp for.
